ibm/resource_ibm_container_vpc_cluster.go

```diff
@@ -501,10 +499,99 @@ func resourceIBMContainerVpcClusterUpdate(d *schema.ResourceData, meta interface
 				"Error updating the worker_count %d: %s", count, err)
 		}
 	}
+	if d.HasChange("zones") {
+		oldZones, newZones := d.GetChange("zones")
+		oldZone := oldZones.([]interface{})
+		newZone := newZones.([]interface{})
+		if len(newZone) > len(oldZone) {
+			for _, nZ := range newZone {
+				newPack := nZ.(map[string]interface{})
+				exists := false
+				for _, oZ := range oldZone {
+					oldPack := oZ.(map[string]interface{})
+					if (strings.Compare(newPack["name"].(string), oldPack["name"].(string)) == 0) && (strings.Compare(newPack["subnet_id"].(string), oldPack["subnet_id"].(string)) == 0) {
+						exists = true
+					}
+				}
+				if !exists {
+					zoneParam := v2.WorkerPoolZone{
+						Cluster:      clusterID,
+						Id:           newPack["name"].(string),
+						SubnetID:     newPack["subnet_id"].(string),
+						WorkerPoolID: "default",
+					}
+					err = csClient.WorkerPools().CreateWorkerPoolZone(zoneParam, targetEnv)
+					if err != nil {
+						return fmt.Errorf("Error adding zone to conatiner vpc cluster: %s", err)
+					}
```

Comment:
Add wait logic to even wait for zone and workers to be ready..Use existing WaitForWorkerPoolAvailable